src/main/java/dev/jbang/ConfigUtil.java

```diff
@@ -0,0 +1,149 @@
+package dev.jbang;
+
+import static dev.jbang.Util.verboseMsg;
+import static dev.jbang.Util.warnMsg;
+
+import java.io.IOException;
+import java.io.Reader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Objects;
+
+import com.google.gson.Gson;
+
+public class ConfigUtil {
+
+	public static class Config extends HashMap<String, Object> {
+		public String editor() {
+			return getString("editor", null);
+		}
+
+		private String getString(String key, String defaultValue) {
+			if (containsKey(key)) {
+				return Objects.toString(get(key));
+			} else {
+				return defaultValue;
+			}
+		}
+	}
+
+	public static final Config defaults;
+
+	static {
+		defaults = new Config();
+	}
+
+	/**
+	 * Returns a Config containing all the settings from local config files merged
+	 * into one. This follows the system where settings that are "nearest" have
+	 * priority. The Config starts out with all the values from `defaults`, any
+	 * values read from files have priority.
+	 *
+	 * @param cwd The current working directory
+	 * @return a Config object
+	 */
+	public static Config getMergedConfig(Path cwd) {
```

Comment:
the docs says files as in possibly more than 2 ? 

isn't the normal behaviour that you get max 2, as in nearest local + global or are intending on capturing and merging all potential parents recursively ?

Reply:
It has the same behavior as aliases, so yes it recurses up the file system tree.